nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,58 @@
+const express = require("express");
+const router = express.Router();
+const meals = require("./../data/meals.json");
+
+/* Respond with the json for all the meals */
+/* router.get("/", async (req, res) => {
+  try {
+    res.json(meals);
+  } catch (error) {
+    throw error;
+  }
+});  */
+
+/* Respond with the json for the meal with the corresponding id */
+router.get("/:id", async (req, res) => {
+  try {
+    const result = getMeal(parseInt(req.params.id));
+    Object.keys(result).length === 0
+      ? res.status(404).json({ error: `${req.params.id} not found` })
+      : res.status(200).json(result);
```

Comment:
It's totally fine to return an error response, however the test expects something else!

Reply:
You will have to apologize me here, but we have not yet learned about proper error handling, so I really do have no idea about how to handle errors here, especially no clue about what the test expect. I could use a bit of guidance here.

Reply:
Maybe I went too far but you seem to catch on really quickly :) So, what I mean for this and my other "test requires" comments is:
There is a test folder under your meal-sharing folder with a parameters.spec.js file. If you run as per the homework instructions the command "npm test" several tests(defined in this parameters.spec.js file) will run to ensure your routes work as expected. For instance, there is a test for this specific route that sends a meal id which is not a number, and expects to get back a 400 response but instead what happens in your case is you "throw new Error()".
What you generally want to do inside the catch blocks is something like:
```
res.status(400).json(error.toString())
```
The reason is, if you "throw" the error and do not set a status for the response, it will never return a response hence if I visit "http://localhost:3000/api/meals/notastringid" it will never load!